trieste/bayesian_optimizer.py

```diff
@@ -999,6 +999,9 @@ def write_summary_observations(
 ) -> None:
     """Write TensorBoard summary for the current step observations."""
     for tag in models:
+        if tag not in tagged_output:
```

Comment:
Very minor point... feel free to ignore. It is probably more intuitive to check against `datasets`, since those concepts exist in multiple places. `datasets` and `tagged_output` keys should match anyway.

```suggestion
        if tag not in datasets:
```